include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1487,10 +1492,13 @@ __remove_elements(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardI
             ::std::forward<_ExecutionPolicy>(__exec), __result, __result + __m,
             [__result, __first, __is_vector](_Tp* __i, _Tp* __j) {
                 __invoke_if_else(
-                    ::std::is_trivial<_Tp>(),
+                    ::std::is_trivially_destructible<_Tp>(),
                     [&]() { __brick_move<_ExecutionPolicy>{}(__i, __j, __first + (__i - __result), __is_vector); },
                     [&]() {
-                        __brick_move_destroy<_ExecutionPolicy>{}(__i, __j, __first + (__i - __result), __is_vector);
+#if __cplusplus >= 201703L
+                        if constexpr (!::std::is_trivially_destructible<_Tp>())
```

Comment:
`if constexpr` is C++17, we still support C++11 for host part.

Reply:
I think we have some macro for using constexpr in cases like this.

Reply:
"if constexpr is C++17, we still support C++11 for host part." - yes, sure.
And it's a reason why I used #if __cplusplus >= 201703L before this constexpr check.

Reply:
> I think we have some macro for using constexpr in cases like this.

I created new define in this file IF_CONSTEXPR_SINCE_CPP17